rtw_excel_report/models/models.py

```diff
@@ -1035,10 +1043,31 @@ def _compute_sale_order_product_detail(self):
                         attr_cfg += ("● " +  cfg.display_name  + ":" + cfg.value  + "\n" )
                         
             attr = attr.rstrip()
-            attr_cfg = attr_cfg.rstrip() 
+            attr_cfg = attr_cfg.rstrip()
             
             line.sale_order_product_detail = attr
             line.sale_order_product_detail_2 = attr_cfg
+            
+    def _compute_sale_order_all_attribute(self):
+        for line in self:
+            attr = ""
+            attributes = line.product_id.product_template_attribute_value_ids  #attr default
+            attributes_cfg=line.config_session_id.custom_value_ids             #attr custom 
+            
+            if attributes:
+                for attribute in attributes:
+                    attr += ("● " + attribute.attribute_id.name + ":" + attribute.product_attribute_value_id.name + "\n" )                    
+                if attributes_cfg:   
+                    for cfg2 in attributes_cfg:
+                        attr += ("● " + cfg2.display_name  + ":" + cfg2.value  + "\n" )
+            else: 
+                if attributes_cfg:   
+                    for cfg2 in attributes_cfg:
+                        attr += ("● " + cfg2.display_name  + ":" + cfg2.value  + "\n" )
+                        
```

Comment:
code này:
                if attributes_cfg:   
                    for cfg2 in attributes_cfg:
                        attr += ("● " + cfg2.display_name  + ":" + cfg2.value  + "\n" )
đang bị duplicate, có thể bỏ ra khỏi if else được

Reply:
ko, sửa như này thì khác j đâu.
ý a là như này:
if attributes:
    for attribute in attributes:
        attr += ("● " + attribute.attribute_id.name + ":" + attribute.product_attribute_value_id.name + "\n" )                    
if attributes_cfg:   
    for cfg2 in attributes_cfg:
        attr += ("● " + cfg2.display_name  + ":" + cfg2.value  + "\n" )